cookies/views.py

```diff
@@ -1180,40 +1180,49 @@ def resource_content(request, resource_id):
         return HttpResponseRedirect(target)
     return HttpResponse('Nope')
 
-def export_coauthor_data(request):
+def export_coauthor_data(request, parent_id):
+    """
+    Exporting coauthor data from a collection detail view
+    """
+
     context = RequestContext(request, {})
 
-    graph = nx.Graph()
+    if parent_id:
+        # Prefetching resources and relations from collection db
+        collection = Collection.objects.prefetch_related('native_resources__relations_from').get(id=parent_id)
```

Comment:
Hint: if no ``Collection`` responds to your query, ``get()`` will throw ``Collection.DoesNotExist``